Task 7.3/Task 7.3/Program.cs

```diff
@@ -0,0 +1,91 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace Task_7._3
+{
+    internal class Program
+    {
+        static void Main(string[] args)
+        {
+            Console.WriteLine("Введите позицию белого короля");
+            var whiteKingPosition = Console.ReadLine();
+
+            if (!IsPositionCorrect(whiteKingPosition))
+            {
+                Console.WriteLine("Некорректная позиция белого короля");
+                Console.ReadKey();
+                return;
+            }
+
+            Console.WriteLine("Введите позицию черного коня");
+            var blackHorsePosition = Console.ReadLine();
+
+            if (!IsPositionCorrect(blackHorsePosition) ||
+                whiteKingPosition == blackHorsePosition ||
+                IsUnderAttack(whiteKingPosition, blackHorsePosition))
+            {
+                Console.WriteLine("Некорректная позиция черного коня или фигуры находятся под боем друг друга");
+                Console.ReadKey();
+                return;
+            }
+
+            Console.WriteLine("Введите предполагаемую позицию хода белого короля");
+            var move = Console.ReadLine();
+
+            if (CanKingMove(move, whiteKingPosition, blackHorsePosition))
+                Console.WriteLine("Ход разрешен");
```

Comment:
Нет проверки на то, что король попадет под бой коня.
```suggestion
            if (CanKingMove(move, whiteKingPosition, blackHorsePosition) &&
                !IsUnderAttack(whiteKingPosition, blackHorsePosition))
                Console.WriteLine("Ход разрешен");
```